src/renderer/i18n/index.js

```diff
@@ -1,31 +1,84 @@
 import Vue from 'vue'
 import VueI18n from 'vue-i18n'
-import yaml from 'js-yaml'
 import fs from 'fs'
 
-const isDev = process.env.NODE_ENV === 'development'
+// List of locales approved for use
+import activeLocales from '../../../static/locales/activeLocales.json'
 
-Vue.use(VueI18n)
+const isDev = process.env.NODE_ENV === 'development'
 
-// List of locales approved for use
-const activeLocales = ['en-US', 'en_GB', 'ar', 'bg', 'ca', 'cs', 'da', 'de-DE', 'el', 'es', 'es_AR', 'es-MX', 'et', 'eu', 'fi', 'fr-FR', 'gl', 'he', 'hu', 'hr', 'id', 'is', 'it', 'ja', 'ko', 'lt', 'nb_NO', 'nl', 'nn', 'pl', 'pt', 'pt-BR', 'pt-PT', 'ro', 'ru', 'sk', 'sl', 'sr', 'sv', 'tr', 'uk', 'vi', 'zh-CN', 'zh-TW']
 const messages = {}
-/* eslint-disable-next-line */
-const fileLocation = isDev ? 'static/locales/' : `${__dirname}/static/locales/`
-
-// Take active locales and load respective YAML file
-activeLocales.forEach((locale) => {
-  try {
-    // File location when running in dev
-    const doc = yaml.load(fs.readFileSync(`${fileLocation}${locale}.yaml`))
-    messages[locale] = doc
-  } catch (e) {
-    console.log(e)
+
+if (isDev) {
+  const { load } = require('js-yaml')
+  // Take active locales and load respective YAML file
+  activeLocales.forEach((locale) => {
+    try {
+      // File location when running in dev
+      const doc = load(fs.readFileSync(`static/locales/${locale}.yaml`))
+      messages[locale] = doc
+    } catch (e) {
+      console.error(e)
+    }
+  })
+}
+
+class CustomVueI18n extends VueI18n {
+  constructor(options) {
+    super(options)
+    this.allLocales = activeLocales
   }
-})
 
-export default new VueI18n({
+  async loadLocale(locale) {
+    // we only lazy load locales for producation builds
+    if (!isDev) {
+      // don't need to load it if it's already loaded
+      if (this.availableLocales.includes(locale)) {
+        return
+      }
+      if (!this.allLocales.includes(locale)) {
+        console.error(`Unable to load unknown locale: "${locale}"`)
+      }
+
+      if (process.env.IS_ELECTRON) {
+        const { brotliDecompressSync } = require('zlib')
+        // locales are only compressed in our production Electron builds
+        try {
+          // decompress brotli compressed json file and then load it
+          // eslint-disable-next-line node/no-path-concat
+          const compressed = fs.readFileSync(`${__dirname}/static/locales/${locale}.json.br`)
+          const data = JSON.parse(brotliDecompressSync(compressed).toString())
+          this.setLocaleMessage(locale, data)
+        } catch (err) {
+          console.error(err)
+        }
+      } else {
+        const url = new URL(window.location.href)
+        url.hash = ''
+        if (url.pathname) {
+          url.pathname += `/static/locales/${locale}.json`
```

Comment:
I noticed that this uses `./static/locales` as the directory, but in `webpack.web.config.js`, the output directory is defined as `./web/static/locales`. 
```javascript
// line 162 of _scripts/webpack.web.config.js
const processLocalesPlugin = new ProcessLocalesPlugin({
  compress: false,
  inputDir: path.join(__dirname, '../static/locales'),
  outputDir: 'web/static/locales',
})
```
I believe this would cause a 404 when fetching for those resources. This could be fixed by making this point to `./web/static/locales` or by making `webpack.web.config.js` point to `./static/locales`.

Reply:
The reason for having a separate subfolder for the web build is so that you can theoretically build both the electron and web builds on the same machine without needing to `yarn clean` between the builds. When deploying the web build you can just copy the contents of the web directory onto the web server, so when it's deployed the "web" directory shouldn't exist anymore.

Also as the index.html file will already be in the web directory adding `/web` to the path in the code would then result in the url to the file being wrong, as it will then be looking for a web directory inside the web directory.

Reply:
I see. I was confused because I was thinking about this wrong. I hadn't realized that index.html would be in `web/`, but that makes complete sense.